File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.java
class Solution {
    public int[] searchRange(int[] nums, int target) {
        int start=findFirst(nums,target);
        int last=findLast(nums,target);
        return new int[]{start,last};
    }
    private int findFirst(int[] nums,int target){
        int low=0;
        int high=nums.length-1;
        int ans=-1;
        while(low<=high){
            int mid=low + (high-low)/2;
            if(target==nums[mid]){
                ans=mid;
                high=mid-1;
            }
            else if(target<nums[mid]){
                high=mid-1;
            }
            else{
                low=mid+1;
            }
        }
        return ans;
    }
    private int findLast(int[] nums,int target){
        int low=0;
        int high=nums.length-1;
        int ans=-1;
        while(low<=high){
            int mid=low+(high-low)/2;
            if(target==nums[mid]){
                ans=mid;
                low=mid+1;
            }
            else if(target>nums[mid]){
                low=mid+1;
            }
            else{
                high=mid-1;
            }
        }
        return ans;
    }
}

File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.py
class Solution:
    def searchRange(self, nums: List[int], target: int) -> List[int]:
        fi=-1
        li=-1
        i=0
        j=len(nums)-1
        m=i+((j-i//2))
        while i<=j:
            if nums[m]==target:
                fi=m
                j=m-1
                m=i+((j-i//2))
            elif nums[m]>target:
                j=m-1
                m=i+((j-i//2))
            else:
                i=m+1
                m=i+((j-i//2))
        i=0
        if(fi!=-1):
            li=fi
            i=fi+1
        j=len(nums)-1
        m=i+((j-i)//2)
        while i<=j:
            if nums[m]==target:
                li=m
                i=m+1
                m=i+((j-i)//2)
            elif nums[m]>target:
                j=m-1
                m=i+((j-i)//2)
            else:
                i=m+1
                m=i+((j-i)//2)
        return [fi,li]